grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +87,65 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // Skip the node as it already been processed by parent node
+        if(isRequiredNode(jcrNode)) {
+            return null;
+        } else {
+            // Build parent node
+            return buildNode(jcrNode, getRequiredChildNodes(jcrNode))
+        }
+    }
+
+    /**
+     * Build node and "only" mandatory child nodes
+     * @param JCR Node
+     * @param Mandatory childNodes
+     * @return Node Proto node
+     */
+    private Node buildNode(JcrNode jcrNode, Collection<JcrNode> childNodes) {
```

Comment:
This is minor .. but the method for marshalling JCR type to Protobuf type was called `toProperty()` below ... could keep the same convention and name this `toNode()` and similarly .. `toProperties()` below instead of `buildProperties()` ...